test/metabase/api/revision_test.clj

```diff
@@ -361,3 +361,35 @@
                  (map #(select-keys % [:description :has_multiple_changes])
                       (mt/user-http-request :crowberto :get 200 "revision" :entity "card" :id card-id))))
           (t2/delete! :model/Card :id card-id))))))
+
+(deftest revert-does-not-create-new-revision
+  (testing "revert a dashboard that previously added cards should not create new revision(#30869)"
```

Comment:
The first time I read this I was a little confused, because reverting a dashboard should create exactly one revision - the one that says "reverted to an earlier version.". I think this should read:
```suggestion
  (testing "revert a dashboard that previously added cards should not recreate duplicate revisions(#30869)"
```